File: Samples/JumpList/cppwinrt/Scenario5_IsSupported.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario5_IsSupported.h"
#include "Scenario5_IsSupported.g.cpp"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::UI::StartScreen;
using namespace Windows::UI::Xaml;
using namespace Windows::UI::Xaml::Navigation;

namespace winrt::SDKTemplate::implementation
{
    Scenario5_IsSupported::Scenario5_IsSupported()
    {
        InitializeComponent();
    }

    void Scenario5_IsSupported::OnNavigatedTo(NavigationEventArgs const&)
    {
        Supported().Text(JumpList::IsSupported() ? L"supports" : L"does not support");
    }
}

File: archived/JumpList/js/js/scenario5-issupported.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    var StartScreen = Windows.UI.StartScreen;
    var JumpList = StartScreen.JumpList;

    var page = WinJS.UI.Pages.define("/html/scenario5-issupported.html", {
        ready: function (element, options) {
            Supported.innerText = JumpList.isSupported() ? "supports" : "does not support";
        }
    });
})();
